### PR TITLE
Added Hackathon, HackathonTeam and HackathonTeamRequest models

### PR DESCRIPTION
**Pull request by mistryharsh28**

### CODE REVIEW

File: hackathon/models.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	team = models.ForeignKey(HackathonTeam, on_delete=CASCADE)
+	message = models.TextField(max_length=500)
+	skills = models.ManyToManyField(Skill)
+	create_date = models.DateField()
+	status_choices = (
+		('A' : 'Accepted'),
+		('P' : 'Pending'),
+		('R' : 'Rejected'),
+		)
+	status = models.CharField(max_length=1, choices=status_choices, default='P')
```
Comment (akksshah) [inline hackathon/models.py]: Why is a Status field required when u already have a status_choices field

File: projectfinder/settings.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'd-m1#4178pi$cof=34ts1q^i%e8f%+omlrk63*2a88b(i0tqah'
```
Comment (akksshah) [inline projectfinder/settings.py]: Secret key must not be visible to anyone directly

File: projectfinder/settings.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+WSGI_APPLICATION = 'projectfinder.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/2.0/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
```
Comment (akksshah) [inline projectfinder/settings.py]: @Kushald2805 @shahp98 What db do we want to use?

File: projectfinder/urls.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+Examples:
+Function views
+    1. Add an import:  from my_app import views
+    2. Add a URL to urlpatterns:  path('', views.home, name='home')
+Class-based views
+    1. Add an import:  from other_app.views import Home
+    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+Including another URLconf
+    1. Import the include() function: from django.urls import include, path
+    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
```
Comment (akksshah) [inline projectfinder/urls.py]: Please remove unnecessary comments

**Comment by shahp98** [comment]
Try removing '--user' from the *pip install* command. Then check why the build fails.

**Comment by shahp98** [comment]
@mistryharsh28 @Kushald2805 
Has the environment been set?

File: hackathon/models.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+class HackathonTeam(models.Model):
+	name = models.CharField(max_length=50, unique=True)
+	leader = models.ForeignKey(User, on_delete=models.CASCADE)
+	current_members = models.ManyToManyField(User)
+	hackathon = models.OneToOneField(Hackathon, on_delete=models.CASCADE)
+	vacancies = models.PositiveSmallIntegerField()
+	closed = models.BooleanField()
+	cut_off_date = models.DateField()
+	skills_required = models.ManyToManyField(Skill)
```
Comment (shahp98) [inline hackathon/models.py]: There is no Skills field yet. Why have you used it here then? Please make the changes.

File: hackathon/models.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+	def __str__(self):
+		return self.name
+
+class HackathonTeamRequest(models.Model):
+	sender = models.ForeignKey(User, on_delete=CASCADE)
+	team = models.ForeignKey(HackathonTeam, on_delete=CASCADE)
+	message = models.TextField(max_length=500)
+	skills = models.ManyToManyField(Skill)
```
Comment (shahp98) [inline hackathon/models.py]: Similar to the previous comment, **Skill** model hasn't be defined yet. Please define it. Only then use its reference.

File: hackathon/models.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	def __str__(self):
+		return self.name
+
+class HackathonTeam(models.Model):
+	name = models.CharField(max_length=50, unique=True)
+	leader = models.ForeignKey(User, on_delete=models.CASCADE)
+	current_members = models.ManyToManyField(User)
+	hackathon = models.OneToOneField(Hackathon, on_delete=models.CASCADE)
+	vacancies = models.PositiveSmallIntegerField()
+	closed = models.BooleanField()
```
Comment (shahp98) [inline hackathon/models.py]: Provide a default.

**Comment by shahp98** [comment]
Opened a new PR for the same.

**Pull request closed without merging**